### ISSUE TITLE
New Rule: require-default-switch-case

### ISSUE THREAD

**Issue by pertrai1** [pertrai1/eslint-plugin-llm-core#45]
## Research Evidence

From empirical LLM bug studies: "Missing Corner Cases" accounts for 15.27% of LLM bugs. LLMs often forget to handle default/else cases.

## Description

Require a `default` case in all switch statements to ensure edge cases are explicitly handled (even if just to throw or no-op with comment).

### Bad
```ts
switch (status) {
  case 'active': return 1;
  case 'inactive': return 0;
  // What about 'pending', 'archived', unknown values?
}
```

### Good
```ts
switch (status) {
  case 'active': return 1;
  case 'inactive': return 0;
  default:
    throw new Error(`Unknown status: ${status}`);
}
```

### Also Good (explicit no-op)
```ts
switch (status) {
  case 'active': return 1;
  case 'inactive': return 0;
  default:
    // Other statuses intentionally ignored
    return -1;
}
```

## Why This Matters for LLMs
- Forces LLMs to think about edge cases
- Makes implicit assumptions explicit
- Catches the "missing corner cases" bug pattern

## Priority
Medium - catches a documented LLM bug pattern
